File: VB/App_Code/EmbedHttpHandler.vb
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Web

Public Class EmbedHttpHandler
	Implements IHttpHandler

	Public ReadOnly Property IsReusable() As Boolean Implements IHttpHandler.IsReusable
		Get
			Return False
		End Get
	End Property

	Public Sub ProcessRequest(ByVal context As HttpContext) Implements IHttpHandler.ProcessRequest
		context.Response.Clear()

		If Not String.IsNullOrEmpty(context.Request.QueryString("flash")) Then
			context.Response.ContentType = "application/x-shockwave-flash"
			context.Response.BinaryWrite(GetByteFile(context.Server.MapPath("~/Files/FlashClocks")))
		End If

		context.Response.End()
	End Sub

	Private Function GetByteFile(ByVal filePath As String) As Byte()
		Dim bytes() As Byte
		Using fileStream As Stream = File.OpenRead(filePath)
			bytes = New Byte(fileStream.Length - 1){}
			For i As Integer = 0 To bytes.Length - 1
				bytes(i) = CByte(fileStream.ReadByte())
			Next i
		End Using

		Return bytes
	End Function
End Class

File: CS/App_Code/EmbedHttpHandler.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Web;

public class EmbedHttpHandler : IHttpHandler {
    public bool IsReusable {
        get { return false; }
    }

    public void ProcessRequest(HttpContext context) {
        context.Response.Clear();
        context.Response.ContentType = "application/x-shockwave-flash";
        context.Response.BinaryWrite(GetByteFile(context.Server.MapPath("~/Files/FlashClocks.swf")));
        context.Response.End();
    }

    public static byte[] GetByteFile(string filePath) {
        using(Stream fileStream = File.OpenRead(filePath)) {
            byte[] bytes = new byte[fileStream.Length];
            fileStream.Read(bytes, 0, (int)fileStream.Length);
            return bytes;
        }
    }
}